inductiva/structures/scenarios/plate_linear_elastic/geometry_utils.py

```diff
@@ -0,0 +1,427 @@
+                                                       zc=0,
+                                                       rx=hole_object.radius,
+                                                       ry=hole_object.radius)
+
+                # Rectangular hole
+                elif hole_object.hole_type == "rectangular":
+                    hole_gmsh = gmsh.model.occ.addRectangle(
+                        x=hole_object.center_x - hole_object.size_x,
+                        y=hole_object.center_y - hole_object.size_y,
+                        z=0,
+                        dx=hole_object.size_x * 2,
+                        dy=hole_object.size_y * 2)
+                    gmsh.model.occ.rotate(dimTags=[(2, hole_gmsh)],
+                                          x=hole_object.center_x,
+                                          y=hole_object.center_y,
+                                          z=0,
+                                          ax=0,
+                                          ay=0,
+                                          az=1,
+                                          angle=math.radians(hole_object.angle))
+
+                # Elliptical hole
+                elif hole_object.hole_type == "elliptical":
+
+                    if hole_object.radius_x > hole_object.radius_y:
+                        hole_gmsh = gmsh.model.occ.addDisk(
+                            xc=hole_object.center_x,
+                            yc=hole_object.center_y,
+                            zc=0,
+                            rx=hole_object.radius_x,
+                            ry=hole_object.radius_y)
+                        gmsh.model.occ.rotate(dimTags=[(2, hole_gmsh)],
+                                              x=hole_object.center_x,
+                                              y=hole_object.center_y,
+                                              z=0,
+                                              ax=0,
+                                              ay=0,
+                                              az=1,
+                                              angle=math.radians(
+                                                  hole_object.angle))
+
+                    else:
+                        hole_gmsh = gmsh.model.occ.addDisk(
+                            xc=hole_object.center_x,
+                            yc=hole_object.center_y,
+                            zc=0,
+                            rx=hole_object.radius_y,
+                            ry=hole_object.radius_x)
+                        gmsh.model.occ.rotate(
+                            dimTags=[(2, hole_gmsh)],
+                            x=hole_object.center_x,
+                            y=hole_object.center_y,
+                            z=0,
+                            ax=0,
+                            ay=0,
+                            az=1,
+                            angle=math.radians(hole_object.angle + 90))
```

Comment:
These should be methods of the hole classes. Then all you'd do would be `hole._to_opencascade_cad()`... (or `_to_occ()` for short).